src/components/ColorPicker/components/ColorPickerItemComponent/ColorPickerItemComponent.jsx

```diff
@@ -22,13 +24,31 @@ const ColorPickerItemComponent = ({
   colorSize,
   tooltipContent,
   isActive,
-  colorShape
+  colorShape,
```

Comment:
it feels like the custom color picker might be better off as a separate component.
This way, we can end up with two "self contained" components (one for regular color and one for custom picker), instead of having one component with many "if/else".

I'm not entirely sure about this suggestion, but I think it's worth considering...